src/pages/Scholarships.jsx

```diff
@@ -26,14 +29,55 @@ function ScholarshipsPage() {
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
+  const [amountFilterVals, setAmountFilterVals] = useState({
+    minAmount: 0,
+    maxAmount: 0,
+  });
+  const location = useLocation();
+  const history = useHistory();
+
+  useEffect(() => {
+    const params = queryString.parse(location.search, { parseNumbers: true });
+    const minParam = params.min ?? 0;
+    const maxParam = params.max ?? 0;
+    setAmountFilterVals({
+      minAmount: minParam > 0 ? minParam : 0,
+      maxAmount: maxParam > 0 ? maxParam : 0,
+    });
+
+    const newParams = { ...params };
+    if (!Number.isInteger(minParam) || minParam <= 0) delete newParams.min;
+    if (!Number.isInteger(maxParam) || maxParam <= 0) delete newParams.max;
+    history.push({ search: queryString.stringify(newParams) });
+  }, [location.search, history]);
 
   useEffect(() => {
     // TODO: Create cancellable promises
     Scholarships.list({ sortField, sortDir })
-      .then(setScholarships)
+      .then((results) => {
+        const filterError =
+          amountFilterVals.maxAmount > 0 &&
+          amountFilterVals.maxAmount <= amountFilterVals.minAmount;
+        if (
+          (amountFilterVals.minAmount === 0 &&
+            amountFilterVals.maxAmount === 0) ||
+          filterError
+        )
+          setScholarships(results);
+        else
+          setScholarships(
+            results.filter((result) =>
+              FilterByAmount(
+                result.data.amount,
+                amountFilterVals.minAmount,
+                amountFilterVals.maxAmount
+              )
+            )
+          );
+      })
```

Comment:
Considering the [min/max constants we set for special values](https://github.com/beyondhb1079/s4us/blob/afa48290e9f1d16397c3a9f1c5bd995e763ec8a0/src/types/ScholarshipAmount.ts#L3-L15), with beforehand param pruning, this and `FilterByAmount` can probably be replaced by:
```js
setScholarships(
  results
    .filter(({data}) => !params.minAmount || data.amount.max <= params.minAmount)
    .filter(({data}) => !params.maxAmount || data.amount.min >= params.maxAmount)
)
```
Note how each filter statement checks if that filter is set, letting all scholarships pass through if the filter isn't set.